tests/schema/extensions/test_max_aliases.py

```diff
@@ -0,0 +1,200 @@
+from typing import Dict, List, Optional, Tuple, Union
+
+from graphql import (
+    GraphQLError,
+    parse,
+    specified_rules,
+    validate,
+)
+
+import strawberry
+from strawberry.extensions.max_aliases import create_validator
+
+
+@strawberry.type
+class Human:
+    name: str
+    email: str
+
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def user(self, name: Optional[str], email: Optional[str]) -> Human:
+        pass
+
+    version: str
+    user1: Human
+    user2: Human
+    user3: Human
+
+
+schema = strawberry.Schema(Query)
+
+
+def run_query(
+    query: str, max_aliases: int
+) -> Tuple[List[GraphQLError], Union[Dict[str, int], None]]:
+    document = parse(query)
+
+    result = None
+
+    validation_rule = create_validator(max_aliases)
+
+    errors = validate(
+        schema._schema,
+        document,
+        rules=(*specified_rules, validation_rule),
+    )
+
+    return errors, result
```

Comment:
I'd test creating a schema with the extension here

something like:


```python
schema = strawberry.Schema(Query, extensions=[...])

return schema.execute_sync(query)
```

so we test the flow end to end :)

Reply:
Added 1 such test.

Do you have some suggestions for manual exploratory testing?

Reply:
oh, I'd do what you've done in `test_works_as_extension` in all the tests. 

I think it is better to test that the extensions+schema work together. So that if we change the internal implementation in future we don't need to change the tests 😊